### ISSUE TITLE
Make build script choose the best possible arch based on the machine without requiring user input 

### ISSUE THREAD

**Issue by nitbhat** [UIUC-PPL/charm#2988]
This was discussed in today's core meeting and is especially relevant for new Charm++ users who are looking to benchmark charm. It is motivated by how software like MPICH and OpenMPI don't require the user to specify any architecture for the build. 

Currently, only `smart-build.pl` script provides this capability but requires the user to answer many questions interactively. Since the user is more likely to use `build` to build Charm++/LIBS, it'll be good to have this capability in that script. 
Based on the current documentation, it is likely that a new user will choose `netlrts` and then have issues related to running the charm program (because of nodelist) or have poor performance since it `netlrts` isn't a performant layer.

**Comment by NK-Nikunj** [comment]
I agree with the points put forward. I'm relatively new to charm++ ecosystem and faced a few problems while trying to run my program on multiple nodes. It will be very nice to have an installation that automatically finds the best configuration according to the system and installs accordingly.
